lib/src/util/csv_parser.dart

```diff
@@ -0,0 +1,78 @@
+/// CSVParser parses a stream of lines into a list of vals compliant with
+/// the CSV standard
+///
+/// See RFC4180 for details on CSV standard
+class CsvParser {
+
+  /// Takes a single line and parses it into a list of values according to the
+  /// csv standard (see RFC4180)
+  static List<String> parseLine(
+      String line) {
+    final records = <String>[];
+    var i = 0;
+    StringBuffer record;
+    while (i < line.length) {
+      record = StringBuffer();
+      if (line[i] == "\"") {
+        // if the csv field begins with a double quote, parse it with
+        // proper character escaping - see RC4180 2.5-2.7
+        i = parseEscapedField(line, record, i);
+      } else {
+        i = parseField(line, record, i);
+      }
+      records.add(record.toString());
+      // increment past the current char (a comma or EOL)
+      i++;
+    }
+    return records;
+  }
+
+  /// Parse and write chars to buff until a comma is reached, then return the
+  /// the index after the last char consumed
+  static int parseField(
+      String line,
+      StringBuffer record,
+      int startIndex) {
+    var i = startIndex;
+    while (i < line.length && line[i] != ",") {
+      if (line[i] == "\"") {
+        throw ArgumentError("A field contained an unescaped double quote. "
```

Comment:
Is argument error the correct error type here? Feels like an error that's only supposed to be understood internall (as does AssertionError). Is there a good error type for user error? Eg something equivalent to `InputError`? We could add one to `exceptions.dart` but using built in exceptions seems strongly preferable especially for generic issues like invalid user input.

Reply:
Maybe [FormatException](https://api.dart.dev/stable/2.4.0/dart-core/FormatException-class.html) would be appropriate here?

Reply:
Done!